pkg/gadget-service/instance-manager/manager.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2023-2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instancemanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime"
+)
+
+type Service interface {
+	GetOperatorMap() map[operators.DataOperator]*params.Params
+}
+
+// Manager manages running gadgets without connection based context. It can run gadgets in the background as
+// well as buffer and multiplex their output.
+type Manager struct {
+	// mu is to be used whenever a gadget is installed or a new client wants to attach to a gadget
+	mu              sync.Mutex
```

Comment:
Sure, it's possible - the reason I'm _not_ doing that here is because it would make that lock essentially "public", while it should be used only internally.